test/metabase/driver/query_processor/expand_resolve_test.clj

```diff
@@ -0,0 +1,286 @@
+                                                        :visibility-type    :normal,
+                                                        :table-id           (id :users)
+                                                        :schema-name        "PUBLIC",
+                                                        :table-name         "USERS",
+                                                        :position           nil,
+                                                        :description        nil,
+                                                        :parent-id          nil,
+                                                        :parent             nil},
+                                                :unit  :year},
+                                  :value       {:value (u/->Timestamp "1980-01-01")
+                                                :field {:field {:field-id           (id :users :last_login)
+                                                                :field-name         "LAST_LOGIN",
+                                                                :field-display-name "Last Login",
+                                                                :base-type          :DateTimeField,
+                                                                :special-type       nil,
+                                                                :visibility-type    :normal,
+                                                                :table-id           (id :users)
+                                                                :schema-name        "PUBLIC",
+                                                                :table-name         "USERS",
+                                                                :position           nil,
+                                                                :description        nil,
+                                                                :parent-id          nil,
+                                                                :parent             nil},
+                                                        :unit  :year}}},
+                   :join-tables  [{:source-field {:field-id   (id :checkins :user_id)
+                                                  :field-name "USER_ID"},
+                                   :pk-field     {:field-id   (id :users :id)
+                                                  :field-name "ID"},
+                                   :table-id     (id :users)
+                                   :table-name   "USERS",
+                                   :schema       "PUBLIC"}]},
+    :fk-field-ids #{(id :checkins :user_id)},
+    :table-ids    #{(id :users)}}]
+  (let [expanded-form (expand/expand {:database (id)
+                                      :type     :query
+                                      :query    {:source-table (id :checkins)
+                                                 :aggregation  ["rows"]
```

Comment:
yeah, we should talk about that.  i'm not in favor of allowing a query to have nothing in the aggregation clause and still be valid.  that clause should be required.